opencog/atoms/base/Link.cc

```diff
@@ -149,3 +149,27 @@ bool Link::operator==(const Atom& other) const
     }
     return true;
 }
+
+bool Link::operator<(const Atom& other) const
+{
+    if (getType() == other.getType()) {
+        const HandleSeq& outgoing = getOutgoingSet();
+        const HandleSeq& other_outgoing = other.getOutgoingSet();
+        Arity arity = outgoing.size();
+        Arity other_arity = other_outgoing.size();
+        if (arity == other_arity) {
+            Arity i = 0;
+            while (i < arity) {
+                Handle ll = outgoing[i];
+                Handle rl = other_outgoing[i];
+                if (ll == rl)
+                    i++;
+                else
+                    return ll->operator<(*rl.atom_ptr());
+            }
+            return false;
+        } else
+            return arity < other_arity;
+    } else
+        return getType() < other.getType();
+}
```

Comment:
This function is the gut of the PR and it doesn't abuse UUID. It doesn't fix anything, it just helps me to reproduce a bug, give me another way and I'm happy to remove that code. Please keep in mind that this code is not used unless DEFAULT_ATOMS_LESS is define as content_based_atoms_less. In other words it doesn't bother anyone.